viewer/viewer.js

```diff
@@ -4969,138 +4973,606 @@ app.post('/removeTags', logAction(), function(req, res) {
-    options.findString = req.body.findString || req.params.findString;
-    pcapSearch(session._id, options, function(err, matched) {
-      cb(null, matched);
-    });
-  },
-  function () {
-    // Check Remotely
-    getViewUrl(fields.node, function(err, viewUrl, client) {
-      var info = url.parse(viewUrl);
-      info.path = Config.basePath(fields.node) + fields.node + "/searchSession/" + session._id;
-      info.agent = (client === http?internals.httpAgent:internals.httpsAgent);
-      addAuth(info, req.user, fields.node);
-      addCaTrust(info, fields.node);
-      var preq = client.request(info, function(pres) {
-        pres.on('end', function () {
-          cb(null, true); // ALW FIX
+    user = user._source;
+
+    molochparser.parser.yy = {
+      emailSearch: user.emailSearch === true,
+      fieldsMap: Config.getFieldsMap()
+    };
+
+    // build session query
+    var query = {
+      from: 0,
+      size: hunt.totalSessions,
+      query: { bool: { filter: [{}] } },
+      _source: ['_id', 'node'],
+      sort: { lastPacket: { order: 'asc' } }
+    };
+
+    // get the size of the query if it is being restarted
+    if (hunt.lastPacketTime) {
+      query.size = hunt.totalSessions - hunt.searchedSessions;
+    }
+
+    if (hunt.query.expression) {
+      try {
+        query.query.bool.filter.push(molochparser.parse(hunt.query.expression));
+      } catch (e) {
+        pauseHuntJobWithError(huntId, hunt, { value: `Couldn't compile hunt query expression: ${e}` });
+        return;
+      }
+    }
+
+    if (user.expression && user.expression.length > 0) {
+      try {
+        // Expression was set by admin, so assume email search ok
+        molochparser.parser.yy.emailSearch = true;
+        var userExpression = molochparser.parse(user.expression);
+        query.query.bool.filter.push(userExpression);
+      } catch (e) {
+        pauseHuntJobWithError(huntId, hunt, { value: `Couldn't compile user forced expression (${user.expression}): ${e}` });
+        return;
+      }
+    }
+
+    lookupQueryItems(query.query.bool.filter, function (lerr) {
+      // TODO Process at most 24 hours?
+      query.query.bool.filter[0] = {
+        range: {
+          lastPacket: {
+            gte: hunt.lastPacketTime || hunt.query.startTime * 1000,
+            lt: hunt.query.stopTime * 1000
+          }
+        }
+      };
+
+      query._source = ['lastPacket', 'node', 'huntId', 'huntName'];
+
+      if (Config.debug > 2) {
+        console.log('HUNT', hunt.name, hunt.userId, '- start:', new Date(hunt.lastPacketTime || hunt.query.startTime * 1000), 'stop:', new Date(hunt.query.stopTime * 1000));
+      }
+
+      // do sessions query
+      Db.search('sessions2-*', 'session', query, function getMoreUntilDone (err, result) {
+        if (err || result.error) {
+          pauseHuntJobWithError(huntId, hunt, { value: `Hunt error searching sessions: ${err}` });
+          return;
+        }
+
+        let hits = result.hits.hits;
```

Comment:
We should check if the results total doesn't match the total we previously had in the hunt, and if they don't match update the hunt so the percent will work right.  (or you can just always update when you update the stats if easier)